### ISSUE TITLE
[BUG]: error: a 'x86_64-linux' with features {} is required

### ISSUE THREAD

**Issue by jiriks74** [catppuccin/nix#437]
Since about 2 weeks ago I'm getting the following error:

```
       error: a 'x86_64-linux' with features {} is required to build '/nix/store/sm69qkm2vngym3snbgwqjdpw8kl5wm6d-catppuccin-install-hook.drv', but I am a 'aarch64-linux' with features {benchmark, big-parallel, nixos-test, uid-range}
```

did something change that now requires `x86_64-linux` specifically?

**Comment by isabelroses** [comment]
This is a known breakage. See https://github.com/catppuccin/nix/pull/384#issuecomment-2564130136

**Comment by isabelroses** [comment]
> May I ask why was that change made? It breaks expected behavior of nix:

We started using more IFD which means that your more likely to come across this breakage, which was previously avoided because of the use of builtin fetchers.

**Comment by getchoo** [comment]
> It breaks expected behavior of nix

This *is* expected behavior of Nix. When you pass the `--all-systems` option, you are evaluating derivations for all systems; sometimes evaluation requires building a derivation for that system first ([IFD](https://nix.dev/manual/nix/2.25/language/import-from-derivation.html)), making it impossible to evaluate said system unless you can build for it

**Comment by jiriks74** [comment]
But it breaks without `--all-systems` as well.

**Comment by budimanjojo** [comment]
This also breaks running `nix flake check` on a x86_64-linux machine to check for aarch64-linux or other architectures and vice versa. I don't know how to fix this other than disabling the module. Here's the log on my Github workflow that's affected if needed: https://github.com/budimanjojo/nix-config/actions/runs/13383711225/job/37376609221

**Comment by budimanjojo** [comment]
> This _is_ expected behavior of Nix. When you pass the `--all-systems` option, you are evaluating derivations for all systems

The problem is it breaks even without `--all-systems`. I think there's something wrong with the implementation because the problem with `nix flake check` was already resolved by this: https://github.com/NixOS/nix/pull/7759.
Maybe somewhere around the logic here? https://github.com/catppuccin/nix/blob/b1ff2a638afa827f1473498190a2c1cae1cf41cf/default.nix#L47

The output that I should get when running `nix flake check` on a x86_64-linux machine with `flake.nix` containing packages for aarch64-linux machine should be this:
```
warning: The check omitted these incompatible systems: aarch64-linux
```

But, when I have `catppuccin.fzf.enable` inside my HM as a Nixos module, I get this `a 'aarch64-linux' with features {} is required to build ...` error.

**Comment by isabelroses** [comment]
> The problem is it breaks even without `--all-systems`. I think there's something wrong with the implementation because the problem with `nix flake check` was already resolved by this: [NixOS/nix#7759](https://github.com/NixOS/nix/pull/7759). Maybe somewhere around the logic here?

The original issue is still open. See https://github.com/NixOS/nix/issues/4265#issuecomment-1565663797 and a potential fix https://github.com/NixOS/nix/issues/4265#issuecomment-2477954746

**Comment by budimanjojo** [comment]
> The original issue is still open. See [NixOS/nix#4265 (comment)](https://github.com/NixOS/nix/issues/4265#issuecomment-1565663797) and a potential fix [NixOS/nix#4265 (comment)](https://github.com/NixOS/nix/issues/4265#issuecomment-2477954746)

Yes I know, but the one I mentioned above fixed everything for me (it skips checking for other architectures like my output above) except for the one with catppuccin.nix. Not sure if there's something that I miss?

**Comment by getchoo** [comment]
> Not sure if there's something that I miss?

The workaround won't apply here as it's for per-system attributes, which NixOS etc. configurations are not. I guess Nix just continues to check nixosConfigurations regardless of their architecture 🤷 

Sadly I don't think there's much we can do on our end aside from just not use IFD, which #392 is tracking. I've added this to in the FAQ in https://github.com/catppuccin/nix/pull/513, so hopefully it will help people come across. Thanks everyone
